src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs

```diff
@@ -131,8 +131,19 @@ PEFile MakePEFile(IAssemblySymbol assembly)
                 // reference assemblies should be fine here, we only need the metadata of references.
                 var reference = _parentCompilation.GetMetadataReference(assembly);
                 Log(CSharpEditorResources.Load_from_0, reference.Display);
-                return TryResolve(reference, PEStreamOptions.PrefetchMetadata)
-                    ?? new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
+
+                var result = TryResolve(reference, PEStreamOptions.PrefetchMetadata);
```

Comment:
`TryResolve` only touches `_inMemoryImagesForTesting`. I'm surprised this method is called in production code and isn't placed in `TestAccessor`

Anything I'm missing?

Reply:
`Resolve` is called by ILSpy, not by us directly, so a test accessor doesn't work.

Reply:
@davidwengier I mean when `TryResolve` is called, it will always return a null in production:

https://github.com/dotnet/roslyn/blob/60b1aef3848ca4789812bc390416339af1a35b1a/src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs#L63-L71

So this call doesn't seem to be anything useful (except *probably* in tests - in which case I think tests should deal with `TestAccessor`).

Changes here (if needed) are out of scope for this PR for sure. I just commented since it caught my eyes that it *seems* to always return null (except in tests).

Reply:
Yeah, I got what you meant. The tests use the test accessor to set up the in memory images cache, and clear, but the calls to actually resolve the reference come from ILSpy. ie, the tests call the decompiled source service, the service calls ILSpy, ILSpy calls Resolve, Resolve calls TryResolve.

Reply:
Got it. Thanks a lot for clarifying @davidwengier. That makes perfect sense.